velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -175,5 +175,39 @@ TEST_F(DateTimeFunctionsTest, makeDate) {
   EXPECT_EQ(makeDate(2023, 3, 29), expectedDate);
 }
 
+TEST_F(DateTimeFunctionsTest, lastDay) {
+  const auto lastDayFunc = [&](const std::optional<Date> date) {
+    return evaluateOnce<Date>("last_day(c0)", date);
+  };
+
+  const auto lastDay = [&](const std::string& dateStr) {
+    Date d0;
+    parseTo(dateStr, d0);
+    return lastDayFunc(d0);
+  };
+
+  auto checkEvaluation = [&](const auto& inputDate, const auto& expectedDate) {
+    Date d0;
+    parseTo(expectedDate, d0);
+    EXPECT_EQ(inputDate, d0);
+  };
+
+  checkEvaluation(lastDay("2015-02-28"), "2015-02-28");
```

Comment:
The key to good error message is to use the EXPECT_EQ macro on the line that is failing, e.g. here

```
EXPECT_EQ(lastDay("2015-02-28"), "2015-02-28");
```

You may need to change lastDay to return date formatted as string or add function to convert string to Date.

Reply:
cc @mbasmanova